apps/web-client/src/routes/orders/customers/[id]/+page.svelte

```diff
@@ -113,6 +145,31 @@
 		await markCustomerOrderLinesAsCollected(db, [id]);
 	};
 
+	// #region book-form
+	let bookFormData = null;
+
+	const onUpdated: SuperForm<BookFormSchema>["options"]["onUpdated"] = async ({ form }) => {
+		/**
+		 * This is a quick fix for `form.data` having all optional properties
+		 *
+		 * Unforuntately, Zod will not infer the correct `data` type from our schema unless we configure `strictNullChecks: true` in our TS config.
+		 * Doing so however raises a mountain of "... potentially undefined" type errors throughout the codebase. It will take a significant amount of work
+		 * to fix these properly.
+		 *
+		 * It is still safe to assume that the required properties of BookEntry are there, as the relative form controls are required
+		 */
+		const data = form?.data as BookEntry;
+
+		try {
+			await upsertBook(db, data);
+			bookFormData = null;
+			customerOrderLines = await getCustomerOrderLines(db, Number(customerId));
+			open.set(false);
```

Comment:
This shouldn't be set explicitly: instead, it should:
- `upsertBook`
- close the form

What should happen after that (as a reactive effect):
- the update to `books` table should trigger the `invalidate` function - invalidating the load function
- the load function runs again (having been invalidated) and retrieves the up-to-date book data

I just checked the reactivity subscriptions and they lack `book` table subscription (see: lines 64-79, `onMount` block)
More on the load function invalidation: https://svelte.dev/tutorial/kit/invalidation